src/robusta/integrations/discord/sender.py

```diff
@@ -0,0 +1,281 @@
+        discord_msg = {
+            "username": "Robusta",
+            "avatar_url": ROBUSTA_LOGO_URL,
+            "embeds": [
+                *[_add_color_to_block(block, msg_color) for block in discord_blocks if "description" in block]
+            ],
+            **_add_color_to_block(header_block, msg_color)
+        }
+        if fields:
+            discord_msg['embeds'].append({"fields": fields, "color": msg_color})
+        return discord_msg
+
+    def __to_discord_diff(
+            self, block: KubernetesDiffBlock, sink_name: str
+    ) -> List[DiscordBlock]:
+        # this can happen when a block.old=None or block.new=None - e.g. the resource was added or deleted
+        if not block.diffs:
+            return []
+
+        _blocks = []
+        _blocks.extend(
+            self.__to_discord(
+                ListBlock(
+                    [
+                        f"*{d.formatted_path}*: {d.other_value} :arrow_right: {d.value}"
+                        for d in block.diffs
+                    ]
+                ),
+                sink_name,
+            )
+        )
+
+        return _blocks
+
+    def __to_discord(self, block: BaseBlock, sink_name: str) -> List[Union[DiscordBlock, Tuple]]:
+        if isinstance(block, MarkdownBlock):
+            if not block.text:
+                return []
+            name, value = self.__extract_markdown_name(block)
+            return [
+                {
+                    "name": name or BLANK_CHAR,
+                    "value": self.apply_length_limit(value) or BLANK_CHAR
+                }
+            ]
+        elif isinstance(block, FileBlock):
+            return [(block.filename, (block.filename, block.contents))]
+        elif isinstance(block, DiscordFieldBlock):
+            return [
+                {
+                    "name": block.name,
+                    "value": block.value,
+                    'inline': block.inline
+                }
+            ]
+        elif isinstance(block, HeaderBlock):
+            return [
+                {
+                    "content": self.apply_length_limit(block.text, 150),
+                }
+            ]
+        elif isinstance(block, DiscordDescriptionBlock):
+            return [
+                {
+                    "description": self.apply_length_limit(block.description),
+                }
+            ]
+        elif isinstance(block, TableBlock):
+            return self.__to_discord(
+                DiscordFieldBlock(
+                    name=block.table_name,
+                    value=block.to_markdown(max_chars=MAX_BLOCK_CHARS, apply_table_name=False).text
+                ), sink_name
+            )
+        elif isinstance(block, ListBlock):
+            return self.__to_discord(block.to_markdown(), sink_name)
+        elif isinstance(block, KubernetesDiffBlock):
+            return self.__to_discord_diff(block, sink_name)
+        else:
+            logging.error(
```

Comment:
I would change it to warning.
There are few blocks that we don't support here, but can still be used, like JsonBlock and CallbackBlock